frontend/src/pages/ProfileRead.tsx

```diff
@@ -0,0 +1,345 @@
+                                    </TableBody>
+                                    <TableBody>
+                                        <TableRow>
+                                            <TableCell style={{ width: "40%" }}>Uplink MBR</TableCell>
+                                            <TableCell>{qosFlow(flowKey, dnn, flow.qosRef!)?.mbrUL}</TableCell>
+                                        </TableRow>
+                                    </TableBody>
+                                    <TableBody>
+                                        <TableRow>
+                                            <TableCell style={{ width: "40%" }}>Downlink MBR</TableCell>
+                                            <TableCell>{qosFlow(flowKey, dnn, flow.qosRef!)?.mbrDL}</TableCell>
+                                        </TableRow>
+                                    </TableBody>
+                                    <TableBody>
+                                        <TableRow>
+                                            <TableCell style={{ width: "40%" }}>Charging Characteristics</TableCell>
+                                            <TableCell>{chargingConfig(dnn, snssai!, flow.filter)}</TableCell>
+                                        </TableRow>
+                                    </TableBody>
+                                </Table>
+                            </Card>
+                        </Box>
+                    </div>
+                ),
+            );
+        }
+        return <div></div>;
+    };
+
+    const upSecurity = (dnn: DnnConfiguration | undefined) => {
+        if (dnn !== undefined && dnn!.upSecurity !== undefined) {
+            const security = dnn!.upSecurity!;
+            return (
+                <div key={security.upIntegr}>
+                    <Box sx={{ m: 2 }}>
+                        <h4>UP Security</h4>
+                        <Card variant="outlined">
+                            <Table>
+                                <TableBody>
+                                    <TableRow>
+                                        <TableCell style={{ width: "40%" }}>Integrity of UP Security</TableCell>
+                                        <TableCell>{security.upIntegr}</TableCell>
+                                    </TableRow>
+                                </TableBody>
+                                <TableBody>
+                                    <TableRow>
+                                            <TableCell style={{ width: "40%" }}>Confidentiality of UP Security</TableCell>
+                                            <TableCell>{security.upConfid}</TableCell>
+                                    </TableRow>
+                                </TableBody>
+                            </Table>
+                        </Card>
+                    </Box>
+                </div>
+            );
+        } else {
+            return <div></div>;
+        }
+    };
```

Comment:
Declare component outside of this component. This will cause lots of object re-creation on every render.